packages/apps/slack/src/services/slack.ts

```diff
@@ -0,0 +1,166 @@
+        for(const conversation of conversations){
+          const thread = await this.fetchThreadForConversation(channel.id,conversation.ts,config)
+
+          const sections: Section[] = thread.map((message, index) => ({
+            offset: index,
+            content: message.text,
+            link: `https://slack.com/api/conversations.replies?channel_id=${channel.id}&ts=${conversation.ts}` 
+          }));
+          const threadDoc : IndexableDocument = {
+            id:conversation.ts, // conversation id
+            organisationId:org.id,
+            source:AppNameDefinitions.SLACK,
+            title:conversation.text, // the main message which lead to conversation
+            metadata:{channel_id:channel.id}, // passing channel id just for top down reference
+            sections:sections, // an array of messages in the specific conversation
+            type: DocType.TEXT,
+            updatedAt: new Date(Date.now())
+          }
+          documents.push(threadDoc);
+          if (documents.length >= 100) {
+            yield documents;
+            documents = [];
+          }
+        }
+      }
+      yield documents;
+      await this.oauthService_.update(oauth.id, {
+        last_sync: new Date(),
+      });
+    } catch (error) {
+      if (error.response && error.response.status === 401) {
+        // Check if it's an unauthorized error
+        this.logger_.info(`Refreshing Slack token for ${org.id} organisation`);
+
+        // Refresh the token
+        const oauthToken = await this.container_["slackOauth"].refreshToken(
+          oauth.refresh_token
+        );
+
+        // Update the OAuth record with the new token
+        await this.oauthService_.update(oauth.id, oauthToken);
+
+        // Retry the request
+        return this.getSlackChannelsAndConversations(org);
+      } else {
+        console.error(error);
+      }
+    }
+
+    this.logger_.info(`Finished oculation of Slack for ${org.id} organisation`);
+  }
+
+  async fetchSlackChannels(config:Config){
+    try{
+      const response = await axios.get(
+        "https://slack.com/api/conversations.list",
+        config
+      )
+
+      if (!response.data)  {
+        return [];
+      }
+
+      const channels = response.data.channels
+      return channels
+    }catch(error){
+      this.logger_.error(
+        "Error fetching Slack Channels in fetchSlackChannels:",
+        error
+      );
+      throw error;
+    }
+  }
+
+  async fetchChannelConversations(channelID:string ,config:Config){
+    try{
+      const conversationsEndpoint = `https://slack.com/api/conversations.history?channel=${channelID}`
+      const response = await axios.get(conversationsEndpoint,config)
+      const conversationsArray = response.data.messages || []
+      const conversations = conversationsArray.map((conversations)=>({
+        id: conversations.ts,
+        text:conversations.text,
+        user: conversations.user
+      }))
```

Comment:
We are assigning `conversations.ts` to `id`, not to `ts`

Reply:
Oh right! Missed that!

Reply:
Could you please create a PR with the change? Thanks for finding the issue